test/integration/targets/yum_repository/tasks/yum_repository.yml

```diff
@@ -0,0 +1,118 @@
+---
+
+- name: disable epel
+  yum_repository:
+    name: epel
+    description: EPEL yum repo
+    baseurl: https://download.fedoraproject.org/pub/epel/{{ ansible_distribution_major_version }}/{{ ansible_architecture }}/
+    state: absent
+
+- name: disable epel (Idempotant)
+  yum_repository:
+    name: epel
+    description: EPEL yum repo
+    baseurl: https://download.fedoraproject.org/pub/epel/{{ ansible_distribution_major_version }}/{{ ansible_architecture }}/
+    state: absent
+  register: epel_remove
+
+- name: check Idempotant
+  assert:
+    that: not epel_remove.changed
+
+- name: install sl
+  yum:
+    name: sl
+    state: present
+  ignore_errors: yes
+  register: sl_result
+
+- debug: var=sl_result
+
+- name: check that install failed
+  assert:
+    that:
+      - sl_result.failed
+      - "sl_result.msg==\"No package matching 'sl' found available, installed or updated\""
+
+- name: re-add epel
+  yum_repository:
+    name: epel
+    description: EPEL yum repo
+    baseurl: https://download.fedoraproject.org/pub/epel/{{ ansible_distribution_major_version }}/{{ ansible_architecture }}/
+    state: present
+  register: epel_add
+
+- name: get repolist
+  shell: yum repolist
+  register: repolist
+
+- name: ensure epel was added
+  assert:
+    that:
+      - "\"epel\" in repolist.stdout"
+      - epel_add.changed
+
+- name: install sl
+  yum:
+    name: sl
+    state: present
+  register: sl_result
+
+- name: check that sl was successfully installed
+  assert:
+    that:
+      - sl_result.changed
+
+- name: remove sl
+  yum:
+    name: sl
+    state: absent
+
+- name: change configuration of epel repo
+  yum_repository:
+    name: epel
+    baseurl: https://download.fedoraproject.org/pub/epel/{{ ansible_distribution_major_version }}/{{ ansible_architecture }}/
+    description: New description
+    async: no
+    enablegroups: no
+    file: epel2
+    exclude:
+      - sl
+    ip_resolve: 4
+    keepalive: no
+  register: epel_add
+
+- name: get epel config file
+  shell: "cat /etc/yum.repos.d/epel2.repo"
+  register: repofile
+
+- debug: var=repofile
+
+- name: check that options are correctly getting written to the repo file
+  assert:
+    that:
+      - "\"async = 0\" in repofile.stdout"
+      - "\"name = New description\" in repofile.stdout"
+      - "\"enablegroups = 0\" in repofile.stdout"
+      - "\"exclude = ['sl']\" in repofile.stdout"
+      - "\"ip_resolve = 4\" in repofile.stdout"
+      - "\"keepalive = 0\" in repofile.stdout"
+
```

Comment:
I wonder if it might be worth re running `name: change configuration of epel repo` and checking it's idempotent?